tests/test_bump_create_commit_message.py

```diff
@@ -55,7 +56,10 @@ def test_bump_pre_commit_changelog(tmp_commitizen_project, mocker, freezer, retr
             """
         )
         cmd.run("git add -A")
-        cmd.run("git commit -m 'fix: _test'")
+        if os.name == "nt":
```

Comment:
these are the same, for non-nt OS this one should work

Reply:
This was failing in CI on Windows only. If you look at the `pytest_report` (attached to this PR), these were the tests failing with the error message

```
commitizen.exceptions.GitCommandError: fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
```

Specifically,

- `test_bump_pre_commit_changelog`
- `test_bump_pre_commit_changelog_fails_always`

This signaled to me no commit had been made, so I captured `sys.out` on the tests and indeed found that to be the case. This is easily reproducible in Command Prompt:

![image](https://user-images.githubusercontent.com/59346180/197348149-f339f46e-4ba9-49f7-808e-2d2c9f2e2db5.png)

In my commit message on `9d40`, I explain what's happening:

> On Windows, `Popen` with `shell=True` reads the `COMSPEC` environment
> variable to determine the shell to run, which defaults to Command
> Prompt (see
> https://docs.python.org/3/library/subprocess.html#subprocess.run).
> For backwards-compatibility for older OS utility scripts, `COMSPEC`
> should not be changed on Windows.
> 
> Command Prompt requires double-quotes instead of single-quotes. See
> 
>   - https://stackoverflow.com/questions/25534995/escape-double-quotes-in-git-config-from-cmd)
>   - https://docs.python.org/3/library/subprocess.html#converting-an-argument-sequence-to-a-string-on-windows
> 
> Note that although the commitizen shell `scripts` shebangs have
> been changed to a portable variety (i.e. `#!/usr/bin/env sh`),
> which runs Git Bash on Windows when Git is installed (i.e.
> the `sh.exe` executable that ships with Git for Windows),
> `subprocess.Popen()` opens a child process from the active
> running shell (i.e. via `CreateProcess()`). Hence, it will
> still read `COMSPEC` and run a Command Prompt shell.

Reply:
But can linux run the windows one?